Gems/GradientSignal/Code/Tests/EditorGradientSignalBakerTests.cpp

```diff
@@ -0,0 +1,244 @@
+            ASSERT_NE(imageInput, nullptr);
+
+            // Make sure the image that was loaded had no errors
+            EXPECT_EQ(imageInput->has_error(), false);
+
+            // Make sure the expected image resolution matches the resolution spec of the actual file that was baked
+            const OIIO::ImageSpec& spec = imageInput->spec();
+            const int imageResolutionX = aznumeric_cast<int>(configuration.m_outputResolution.GetX());
+            const int imageResolutionY = aznumeric_cast<int>(configuration.m_outputResolution.GetY());
+            EXPECT_EQ(spec.width, imageResolutionX);
+            EXPECT_EQ(spec.height, imageResolutionY);
+
+            // Read in the image pixels
+            AZStd::vector<float> pixels(spec.width * spec.height * spec.nchannels);
+            imageInput->read_image(OIIO::TypeDesc::FLOAT, pixels.data());
+
+            // For most tests we are going to check against the first pixel we find, but for the bounds overlap
+            // test case we need to instead compare against the opposite edge of the image
+            int firstPixelIndex = 0;
+            if (!compareAgainstFirstPixel)
+            {
+                firstPixelIndex = (spec.width * spec.nchannels) - 1;
+            }
+
+            // For the R8 output format, we don't have enough granularity to satisfy the default
+            // float value tolerance, so we need to calculate the actual tolerance threshold
+            float tolerance = AZ::Constants::Tolerance;
+            if (outputFormat == GradientSignal::OutputFormat::R8)
+            {
+                tolerance = 1.0f / std::numeric_limits<AZ::u8>::max();
+            }
+            EXPECT_TRUE(AZ::IsClose(pixels[firstPixelIndex], expectedValue, tolerance));
+
+            // For the bounds overlap test case, we need to verify the first pixel (0,0) is outside
+            // the bounds so it will be 0.0f
+            if (!compareAgainstFirstPixel)
+            {
+                EXPECT_TRUE(AZ::IsClose(pixels[0], 0.0f, tolerance));
+            }
+
+            imageInput->close();
+            delete bakeJob;
+        }
+    };
+
+    TEST_F(EditorGradientSignalBakerTestsFixture, InvalidInputGradient)
+    {
+        // An invalid input gradient should cause the output image to be entirely 0.0f values
+        TestBakeImage(".png", GradientSignal::OutputFormat::R8, AZ::Vector2(10.0f), false);
+    }
+
+    TEST_F(EditorGradientSignalBakerTestsFixture, BoundsHalfOverlap)
+    {
+        // Creating an input bounds that half overlaps our test shape will result in half being the
+        // expected constant value and the other half 0.0f
+        AZ::Aabb inputBounds = AZ::Aabb::CreateFromMinMax(AZ::Vector3(-5.0f, -5.0f, 0.0f), AZ::Vector3(5.0f, 5.0f, 0.0f));
+        TestBakeImage(".png", GradientSignal::OutputFormat::R8, AZ::Vector2(10.0f), true, inputBounds);
+    }
+
+    TEST_F(EditorGradientSignalBakerTestsFixture, BakedImage_PNG_R8)
```

Comment:
Suggestion: could these permutations not be wrapped up in a single [parametized test](https://github.com/google/googletest/blob/main/docs/advanced.md#value-parameterized-tests)?

Reply:
Cool I didn't know about these! I think it might over-complicate it for these particular tests because I'm sharing a single fixture across them all and the negative use-cases use a different number of params than the regular cases, but I'll remember for next time, thanks!